core/src/main/scala/org/apache/spark/util/SerializableConfiguration.scala

```diff
@@ -20,7 +20,13 @@ import java.io.{ObjectInputStream, ObjectOutputStream}
 
 import org.apache.hadoop.conf.Configuration
 
-private[spark]
+import org.apache.spark.annotation.DeveloperApi
+
+/**
+ * Helper wrapper to serialize a Hadoop configuration. Intended for use when implementing
+ * DataSourceV2 readers & writers which depend on the Hadoop configuration from the driver node.
+ */
+@DeveloperApi
```

Comment:
Shall we add `@Unstable` to stay safe, and can you check if this is callable in Java side as well?

Reply:
Sure, I'm fine to put an unstable on this. I'll add a test in Java which calls the constructor on this.

Reply:
This file [hasn't changed in 4 years](https://github.com/apache/spark/commits/master/core/src/main/scala/org/apache/spark/util/SerializableConfiguration.scala), so I don't think there is much risk in using `@DeveloperApi`. I think having a Java test is a good idea though (if it doesn't work, we can replace it with a Java implementation).